File: Selenium/src/test/java/pages/LoginPage.java
package pages;

import org.openqa.selenium.chrome.ChromeDriver;
import util.commonFunctions;
import org.junit.Assert;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;


public class LoginPage {


    private WebDriver driver;

    private By txt_username = By.xpath("//input[@id='user-name']");
    private By txt_password = By.id("password");
    private By login_button = By.id("login-button");
    private By title_txt = By.xpath("//div[@class='app_logo']");
    private By error_text = By.xpath("//h3[@data-test='error']");

    private By burger_menu = By.xpath("//button[@id='react-burger-menu-btn']");
    private By logout_button = By.id("logout_sidebar_link");


    /**
     * Initialise the driver
     * @param driver
     */
    public LoginPage(WebDriver driver)
    {
        this.driver = driver;

    }

    public void enterUsername(String username)
    {
        commonFunctions.explicitlywait(driver, driver.findElement(txt_username), 10);
        driver.findElement(txt_username).sendKeys(username);
    }
    public void enterPassword(String password)
    {
        driver.findElement(txt_password).sendKeys(password);
    }

    public void clickLoginbutton()
    {
        driver.findElement(login_button).click();
    }

    public void Verifylogin()
    {
        String actualtxt = driver.findElement(title_txt).getText();
        Assert.assertEquals("Swag Labs", actualtxt);
    }

    public String Verifyinvalidlogin()
    {
        String errortext = driver.findElement(error_text).getText();
        return errortext;
    }

    public void selectBurgerMenu()
    {
        driver.findElement(burger_menu).click();
    }

    public void ClickOnLogout()
    {
        driver.findElement(logout_button).click();
    }

    public boolean validatelogoutsuccess()
    {
        return driver.findElement(login_button).isDisplayed();
    }
}

File: playwright_python/pages/login_page.py
from playwright.sync_api import Page

class LoginPage:
    def __init__(self, page: Page):
        self.page = page
        self.username_input = "#user-name"
        self.password_input = "#password"
        self.login_button = "#login-button"
        self.error_message = "[data-test='error']"
        self.home_logo = ".app_logo"
        
    def load(self):
        self.page.goto("https://www.saucedemo.com/")
        self.page.wait_for_timeout(3000)

    def login(self, username: str, password: str):
        self.page.fill(self.username_input, username)
        self.page.fill(self.password_input, password)
        self.page.wait_for_timeout(1000)
        self.page.click(self.login_button)
        #self.page.wait_for_timeout(3000)

    def is_logged_in(self) -> bool:
        return self.page.is_visible(self.home_logo)

    def login_failed(self) -> bool:
        return self.page.is_visible(self.error_message)
